src/guc.c

```diff
@@ -218,6 +224,94 @@ assign_max_open_chunks_per_insert_hook(int newval, void *extra)
 	validate_chunk_cache_sizes(ts_guc_max_cached_chunks_per_hypertable, newval);
 }
 
+static Oid
+get_segmentby_func(char *input_name)
+{
+	List *namelist = NIL;
+	char *func_name;
+
+	if (strlen(input_name) == 0)
+	{
+		return InvalidOid;
+	}
+
+	func_name = pstrdup(input_name);
+#if PG16_LT
+	namelist = stringToQualifiedNameList(func_name);
+#else
+	namelist = stringToQualifiedNameList(func_name, NULL);
+#endif
```

Comment:
AFAICT, there is no need to make a copy here since `stringToQualifiedNameList` deals with this.
```suggestion
#if PG16_LT
	namelist = stringToQualifiedNameList(input_name);
#else
	namelist = stringToQualifiedNameList(input_name, NULL);
#endif
```